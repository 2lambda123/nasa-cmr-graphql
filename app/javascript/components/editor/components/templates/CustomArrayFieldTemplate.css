.element {
  border-left: solid 6px rgb(240, 240, 240);
  border-bottom: solid 2px rgb(240, 240, 240);
  margin-bottom: 40px;
  padding-left: 10px;
  /* background-color: rgb(240, 240, 240); */
}

.remove-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-right: -12px;
  padding-bottom: 10px
}

.remove-title {
  height: 14px
}

.remove-label {
  color: red;
  font-weight: 'bold'
}

.remove-button {
  color: red
}

.add-header {
  margin-top: -30px
}

.add-button {
  color: rgb(31, 107, 162);
  margin-left: -12px;
}

.add-label {
  color: rgb(31, 107, 162);
  font-weight: bold
}