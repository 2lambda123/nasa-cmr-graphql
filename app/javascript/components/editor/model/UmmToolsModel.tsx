/* eslint-disable dot-notation */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { UiSchema } from '@rjsf/utils'
import toolInformationUiSchema from '../data/configuration/uiSchemas/tools/tool_information'
import relatedUrlsUiSchema from '../data/configuration/uiSchemas/tools/related_urls'
import toolsConfiguration from '../data/configuration/uiForms/toolsConfiguration'
import organizationUiSchema from '../data/configuration/uiSchemas/tools/organization'
import UmmModel from './UmmModel'
import descriptiveKeywordsUiSchema from '../data/configuration/uiSchemas/tools/descriptive_keywords'
import potentialActionUiSchema from '../data/configuration/uiSchemas/tools/potential_action'
import toolContactsUISchema from '../data/configuration/uiSchemas/tools/tool_contacts'
import compatibilityAndUsabilityUiSchema from '../data/configuration/uiSchemas/tools/compatibility_and_usability'
import CustomArrayFieldTemplate from '../components/templates/CustomArrayFieldTemplate'
import schema from '../data/schemas/umm_tools_schema'

export default class UmmToolsModel extends UmmModel {
  constructor() {
    super(schema)
  }
  get documentTypeForDisplay() {
    return 'Tool Record'
  }
  get documentType() {
    return 'tool_drafts'
  }
  get formSections() {
    return toolsConfiguration
  }
  getFormSchema(): { [key: string]: object } {
    // use this function to edit schema
    return super.getFormSchema()
  }
  get uiSchema() {
    const base: UiSchema = {
      'ui:submitButtonOptions': { norender: true, submitText: 'Save' },
      'ui:ArrayFieldTemplate': CustomArrayFieldTemplate
    }
    if (this.currentSection.displayName === 'Related URLs') {
      const uiSchema: any = relatedUrlsUiSchema
      const urlUiSchema: any = uiSchema.RelatedURLs.items
      urlUiSchema['ui:service'] = this.service

      // Format
      const formatUiSchema: any = uiSchema.RelatedURLs.items.Format
      formatUiSchema['ui:service'] = this.service
      formatUiSchema['ui:keyword_scheme'] = 'granule_data_format'
      formatUiSchema['ui:keyword_scheme_column_names'] = ['short_name']

      // Mime Type
      const mimeTypeUiSchema: any = uiSchema.RelatedURLs.items.MimeType
      mimeTypeUiSchema['ui:service'] = this.service
      mimeTypeUiSchema['ui:keyword_scheme'] = 'mime_type'
      mimeTypeUiSchema['ui:keyword_scheme_column_names'] = ['mime_type']
      return { ...uiSchema, ...base }
    }

    if (this.currentSection.displayName === 'Tool Information') {
      const uiSchema: any = toolInformationUiSchema
      const urlUiSchema: any = uiSchema.URL
      urlUiSchema['ui:service'] = this.service
      urlUiSchema['ui:keyword_scheme'] = 'rucontenttype'
      return { ...uiSchema, ...base }
    }
    if (this.currentSection.displayName === 'Descriptive Keywords') {
      const uiSchema = descriptiveKeywordsUiSchema
      const toolKeywords: any = uiSchema.ToolKeywords
      toolKeywords['ui:service'] = this.service

      toolKeywords['ui:keyword_scheme'] = 'science_keywords'
      toolKeywords['ui:picker_title'] = 'TOOL KEYWORD'
      toolKeywords['ui:scheme_values'] = [
        'ToolCategory', 'ToolTopic', 'ToolTerm', 'ToolSpecificTerm'
      ]
      toolKeywords['ui:keyword_scheme_column_names'] = ['toolkeywords', 'category', 'topic', 'term', 'variable_level_1', 'variable_level_2', 'variable_level_3']
      toolKeywords['ui:filter'] = (path: string[]) => path[1] !== 'EARTH SCIENCE'

      return { ...uiSchema, ...base }
    }

    if (this.currentSection.displayName === 'Tool Organizations') {
      const uiSchema: any = organizationUiSchema
      const organizations: any = uiSchema.Organizations.items
      organizations['ui:service'] = this.service
      organizations['ui:keyword_scheme'] = 'providers'
      organizations['ui:keyword_scheme_column_names'] = ['short_name', 'long_name', 'url']
      return { ...uiSchema, ...base }
    }

    if (this.currentSection.displayName === 'Tool Contacts') {
      const uiSchema: any = toolContactsUISchema
      return { ...uiSchema, ...base }
    }

    if (this.currentSection.displayName === 'Potential Action') {
      const uiSchema: any = potentialActionUiSchema
      return { ...uiSchema, ...base }
    }
    if (this.currentSection.displayName === 'Compatibility And Usability') {
      const uiSchema: any = compatibilityAndUsabilityUiSchema
      return { ...uiSchema, ...base }
    }
    return { ...base }
  }

  // In this case the old preview sends the same section name that the react component uses (just lowercase), so
  // nothing needs to be migrated. once the old MMT preview is replaced, this 'migrate" functionality can be removed.
  migratedSectionName(sectionName: string): string {
    return sectionName
  }
}
