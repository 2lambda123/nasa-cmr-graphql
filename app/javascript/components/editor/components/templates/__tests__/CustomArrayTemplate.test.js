import Form from '@rjsf/bootstrap-4'
import React from 'react'
import {
  render, fireEvent, screen, waitFor
} from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import validator from '@rjsf/validator-ajv8'
import UmmToolsModel from '../../../model/UmmToolsModel'
import MetadataEditor from '../../../MetadataEditor'
import CustomArrayFieldTemplate from '../CustomArrayFieldTemplate'

const model = new UmmToolsModel()
const editor = new MetadataEditor(model)
const schema = {
  title: 'CustomArrayFieldTest',
  type: 'array',
  items: {
    properties: {
      firstName: {
        title: 'First Name',
        type: 'string'
      },
      lastName: {
        title: 'Last Name',
        type: 'string'
      },
      age: {
        title: 'Age',
        type: 'integer'
      }
    }
  }
}

const templates = {
  ArrayFieldTemplate: CustomArrayFieldTemplate
}
jest.useFakeTimers()
describe('Custom Array Template', () => {
  it('renders the custom array template and adds an array', async () => {
    const { getByTestId, container } = render(
      <BrowserRouter>
        <Form validator={validator} schema={schema} templates={templates} formContext={{ editor }} />
      </BrowserRouter>
    )
    const mockSetTimeout = jest.fn()
    const addNewField = getByTestId('custom-array-template-add-btn')
    HTMLElement.prototype.scrollIntoView = jest.fn()

    await waitFor(async () => {
      fireEvent.click(await addNewField)
    })

    expect(screen.getByTestId('custom-array-element')).toHaveTextContent('First Name')
    expect(screen.getByTestId('custom-array-element')).toHaveTextContent('Last Name')
    expect(screen.getByTestId('custom-array-element')).toHaveTextContent('Age')

    const addAnotherNewField = getByTestId('custom-array-template-add-another-btn')
    await waitFor(async () => {
      fireEvent.click(await addAnotherNewField)
    })

    jest.runAllTimers(mockSetTimeout)

    expect(HTMLElement.prototype.scrollIntoView).toBeCalledTimes(2)
    const removeButtons = container.getElementsByClassName('custom-array-template-remove-btn')
    expect(removeButtons).toHaveLength(2)
    jest.runAllTimers(mockSetTimeout)

    expect(container).toMatchSnapshot()
  })
  it('renders the custom array template. 1.) Adds a field 2.) removes the added field', async () => {
    const { getByTestId, container } = render(
      <BrowserRouter>
        <Form validator={validator} schema={schema} templates={templates} formContext={{ editor }} />
      </BrowserRouter>
    )
    const addNewField = getByTestId('custom-array-template-add-btn')
    await waitFor(async () => {
      fireEvent.click(await addNewField)
      expect(screen.getByTestId('custom-array-element')).toHaveTextContent('First Name')
      expect(screen.getByTestId('custom-array-element')).toHaveTextContent('Last Name')
      expect(screen.getByTestId('custom-array-element')).toHaveTextContent('Age')

      const removeField = getByTestId('custom-array-template')

      fireEvent.click(await removeField)
    })

    expect(container).toMatchSnapshot()
  })
  it('does not render a header when hide-header is set to true', async () => {
    const uiSchema = {
      'ui:title': 'Custom Array Field Template',
      'ui:hide-header': true,
      'ui:field': 'layout_grid',
      'ui:layout_grid': {
        'ui:row': [
          {
            'ui:col': {
              md: 12,
              children: [
                {
                  'ui:row': [
                    { 'ui:col': { md: 6, children: ['firstName'] } },
                    { 'ui:col': { md: 6, children: ['lastName'] } },
                    { 'ui:col': { md: 6, children: ['age'] } }
                  ]
                }]
            }
          }
        ]
      }
    }
    const { container } = render(<Form validator={validator} schema={schema} uiSchema={uiSchema} templates={templates} formContext={{ editor }} />)
    expect(screen.getByTestId('custom-array-element--heading')).not.toHaveTextContent('Custom Array Field Template')

    expect(container).toMatchSnapshot()
  })
})
