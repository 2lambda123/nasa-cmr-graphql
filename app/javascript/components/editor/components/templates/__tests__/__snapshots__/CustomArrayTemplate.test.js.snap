// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Array Template renders the custom array template and adds an array 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group field field-array"
    >
      <div
        class="form-group"
      >
        <div
          class="metadata-editor-array-field"
          data-testid="custom-array-template"
        >
          <div
            class="h2-box"
          >
            <span
              class="h2-title"
            >
              Custom Array Field Test
            </span>
          </div>
          <p
            class="description-box"
          />
          <br />
          <div
            class="array-item element"
          >
            <div
              class="custom-array-template-remove-btn"
            >
              <div
                class="remove-header"
              >
                <div
                  class="h3-title"
                >
                  <span>
                    Custom Array Field Test
                  </span>
                  <span>
                     (1 of 2)
                  </span>
                </div>
                <button
                  class="btn"
                  data-testid="custom-array-template__remove-button--custom-array-field-test"
                  type="button"
                >
                  <span
                    class="remove-button"
                  >
                    <i
                      class="fa-solid fa-circle-minus fa-lg"
                    />
                  </span>
                   
                  <span
                    class="remove-label"
                  >
                     
                    Remove
                  </span>
                </button>
              </div>
            </div>
            <div
              class="custom-array-template-element"
            />
            <div
              class="mb-2  d-flex align-items-center row"
            >
              <div
                class="array-item col-lg-12 col-12"
                data-testid="custom-array-element"
              >
                <div
                  class="form-group field field-object"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="p-0 container-fluid"
                    >
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_0_firstName"
                              >
                                First Name
                              </label>
                              <input
                                aria-describedby="root_0_firstName__error root_0_firstName__description root_0_firstName__help"
                                class="form-control"
                                id="root_0_firstName"
                                name="root_0_firstName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_0_lastName"
                              >
                                Last Name
                              </label>
                              <input
                                aria-describedby="root_0_lastName__error root_0_lastName__description root_0_lastName__help"
                                class="form-control"
                                id="root_0_lastName"
                                name="root_0_lastName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-integer"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_0_age"
                              >
                                Age
                              </label>
                              <input
                                aria-describedby="root_0_age__error root_0_age__description root_0_age__help"
                                class="form-control"
                                id="root_0_age"
                                name="root_0_age"
                                placeholder=""
                                step="1"
                                type="number"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="array-item element"
          >
            <div
              class="custom-array-template-remove-btn"
            >
              <div
                class="remove-header"
              >
                <div
                  class="h3-title"
                >
                  <span>
                    Custom Array Field Test
                  </span>
                  <span>
                     (2 of 2)
                  </span>
                </div>
                <button
                  class="btn"
                  data-testid="custom-array-template__remove-button--custom-array-field-test"
                  type="button"
                >
                  <span
                    class="remove-button"
                  >
                    <i
                      class="fa-solid fa-circle-minus fa-lg"
                    />
                  </span>
                   
                  <span
                    class="remove-label"
                  >
                     
                    Remove
                  </span>
                </button>
              </div>
            </div>
            <div
              class="custom-array-template-element"
            />
            <div
              class="mb-2  d-flex align-items-center row"
            >
              <div
                class="array-item col-lg-12 col-12"
                data-testid="custom-array-element"
              >
                <div
                  class="form-group field field-object"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="p-0 container-fluid"
                    >
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_1_firstName"
                              >
                                First Name
                              </label>
                              <input
                                aria-describedby="root_1_firstName__error root_1_firstName__description root_1_firstName__help"
                                class="form-control"
                                id="root_1_firstName"
                                name="root_1_firstName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_1_lastName"
                              >
                                Last Name
                              </label>
                              <input
                                aria-describedby="root_1_lastName__error root_1_lastName__description root_1_lastName__help"
                                class="form-control"
                                id="root_1_lastName"
                                name="root_1_lastName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-integer"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_1_age"
                              >
                                Age
                              </label>
                              <input
                                aria-describedby="root_1_age__error root_1_age__description root_1_age__help"
                                class="form-control"
                                id="root_1_age"
                                name="root_1_age"
                                placeholder=""
                                step="1"
                                type="number"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="custom-array-template-add-btn add-header"
          >
            <button
              class="btn"
              data-testid="custom-array-template__add-button--custom-array-field-test"
              type="button"
            >
              <span
                class="add-button"
              >
                <i
                  class="fa-solid fa-circle-plus fa-lg"
                />
              </span>
               
              <span
                class="add-label"
              >
                <span
                  data-testid="custom-array-template-add-another-btn"
                >
                  Add another
                </span>
                 
                Custom Array Field Test
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`Custom Array Template renders the custom array template. 1.) Adds a field 2.) removes the added field 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group field field-array"
    >
      <div
        class="form-group"
      >
        <div
          class="metadata-editor-array-field"
          data-testid="custom-array-template"
        >
          <div
            class="h2-box"
          >
            <span
              class="h2-title"
            >
              Custom Array Field Test
            </span>
          </div>
          <p
            class="description-box"
          />
          <br />
          <div
            class="array-item element"
          >
            <div
              class="custom-array-template-remove-btn"
            >
              <div
                class="remove-header"
              >
                <div
                  class="h3-title"
                >
                  <span>
                    Custom Array Field Test
                  </span>
                  <span>
                     (1 of 2)
                  </span>
                </div>
                <button
                  class="btn"
                  data-testid="custom-array-template__remove-button--custom-array-field-test"
                  type="button"
                >
                  <span
                    class="remove-button"
                  >
                    <i
                      class="fa-solid fa-circle-minus fa-lg"
                    />
                  </span>
                   
                  <span
                    class="remove-label"
                  >
                     
                    Remove
                  </span>
                </button>
              </div>
            </div>
            <div
              class="custom-array-template-element"
            />
            <div
              class="mb-2  d-flex align-items-center row"
            >
              <div
                class="array-item col-lg-12 col-12"
                data-testid="custom-array-element"
              >
                <div
                  class="form-group field field-object"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="p-0 container-fluid"
                    >
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_0_firstName"
                              >
                                First Name
                              </label>
                              <input
                                aria-describedby="root_0_firstName__error root_0_firstName__description root_0_firstName__help"
                                class="form-control"
                                id="root_0_firstName"
                                name="root_0_firstName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_0_lastName"
                              >
                                Last Name
                              </label>
                              <input
                                aria-describedby="root_0_lastName__error root_0_lastName__description root_0_lastName__help"
                                class="form-control"
                                id="root_0_lastName"
                                name="root_0_lastName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-integer"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_0_age"
                              >
                                Age
                              </label>
                              <input
                                aria-describedby="root_0_age__error root_0_age__description root_0_age__help"
                                class="form-control"
                                id="root_0_age"
                                name="root_0_age"
                                placeholder=""
                                step="1"
                                type="number"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="array-item element"
          >
            <div
              class="custom-array-template-remove-btn"
            >
              <div
                class="remove-header"
              >
                <div
                  class="h3-title"
                >
                  <span>
                    Custom Array Field Test
                  </span>
                  <span>
                     (2 of 2)
                  </span>
                </div>
                <button
                  class="btn"
                  data-testid="custom-array-template__remove-button--custom-array-field-test"
                  type="button"
                >
                  <span
                    class="remove-button"
                  >
                    <i
                      class="fa-solid fa-circle-minus fa-lg"
                    />
                  </span>
                   
                  <span
                    class="remove-label"
                  >
                     
                    Remove
                  </span>
                </button>
              </div>
            </div>
            <div
              class="custom-array-template-element"
            />
            <div
              class="mb-2  d-flex align-items-center row"
            >
              <div
                class="array-item col-lg-12 col-12"
                data-testid="custom-array-element"
              >
                <div
                  class="form-group field field-object"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="p-0 container-fluid"
                    >
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_1_firstName"
                              >
                                First Name
                              </label>
                              <input
                                aria-describedby="root_1_firstName__error root_1_firstName__description root_1_firstName__help"
                                class="form-control"
                                id="root_1_firstName"
                                name="root_1_firstName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-string"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_1_lastName"
                              >
                                Last Name
                              </label>
                              <input
                                aria-describedby="root_1_lastName__error root_1_lastName__description root_1_lastName__help"
                                class="form-control"
                                id="root_1_lastName"
                                name="root_1_lastName"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="row"
                        style="margin-bottom: 10px;"
                      >
                        <div
                          class="col-12"
                        >
                           
                          <div
                            class="form-group field field-integer"
                          >
                            <div
                              class="form-group"
                            >
                              <label
                                class="form-label"
                                for="root_1_age"
                              >
                                Age
                              </label>
                              <input
                                aria-describedby="root_1_age__error root_1_age__description root_1_age__help"
                                class="form-control"
                                id="root_1_age"
                                name="root_1_age"
                                placeholder=""
                                step="1"
                                type="number"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="custom-array-template-add-btn add-header"
          >
            <button
              class="btn"
              data-testid="custom-array-template__add-button--custom-array-field-test"
              type="button"
            >
              <span
                class="add-button"
              >
                <i
                  class="fa-solid fa-circle-plus fa-lg"
                />
              </span>
               
              <span
                class="add-label"
              >
                <span
                  data-testid="custom-array-template-add-another-btn"
                >
                  Add another
                </span>
                 
                Custom Array Field Test
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
